libs/pdi-scanner/src/rust/main.rs

```diff
@@ -218,7 +201,33 @@ fn main() -> color_eyre::Result<()> {
 
     let mut client: Option<Client<Scanner>> = None;
     let mut raw_image_data = RawImageData::new();
-    let mut scan_in_progress = false;
+
+    // We reject sending a command while a scan is in progress because it will
+    // interrupt the scan. To ensure this flag gets reset whenever a scan stops
+    // (whether successfully or with an error or disconnection), we manage this
+    // flag in the send_response/send_event functions, since they are called in
+    // basically every case when the scanner state changes.
+    let scan_in_progress = Cell::new(false);
```

Comment:
I'm not sure if this is the most idiomatic way to implement the functionality I want (factoring out the logic to update this flag into a few different functions), so open to suggestions for other patterns

Reply:
I think this is okay. `Cell` gives us what we need in terms of mutability from multiple closures, so I'd keep this as-is.